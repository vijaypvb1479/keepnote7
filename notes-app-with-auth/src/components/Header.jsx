import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';

import SearchBar from './SearchBar';
import { Button } from '@material-ui/core';
import ResponsiveDrawer from './ResponsiveDrawer';
import { firebase, googleProvider } from '../firebase/firebase';

const styles = theme => ({
  root: {
    width: '100%',
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  title: {
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  grow: {
    flexGrow: 1
  },
});

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoggedIn: localStorage.getItem('isLoggedIn') || false,
    }
    this.handleLogin = this.handleLogin.bind(this);
    this.handleLogout = this.handleLogout.bind(this);
    this.handleNoteCurrentPage = this.handleNoteCurrentPage.bind(this);
    this.handleRemCurrentPage = this.handleRemCurrentPage.bind(this);
  }

  handleLogin() {
    return firebase.auth().signInWithPopup(googleProvider).then(success => {
      var user = firebase.auth().currentUser;
      if (user != null) {
        let email = user.email;
        localStorage.setItem('LoggedInUser', email);
      }
      localStorage.setItem('isLoggedIn', true);
      this.setState({ isLoggedIn: true });
    })

  }
  handleNoteCurrentPage(){ this.props.handleCurrentPage('notes')}
  handleRemCurrentPage(){ this.props.handleCurrentPage('rem')}
  handleLogout() {
   return firebase.auth().signOut().then(() =>  {
      localStorage.removeItem('isLoggedIn', false);
      localStorage.removeItem('LoggedInUser');
      this.setState({ isLoggedIn: false });
    }).catch(function (error) {
      // An error happened.
    });
    
  }

  render() {
    const { classes, handleSearchNote,currentPage } = this.props;
    const { isLoggedIn } = this.state;
    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
            <Typography className={classes.title} variant="h6" color="inherit" noWrap>
              Notes
          </Typography>
            <SearchBar handleSearchNote={handleSearchNote}/>
            <div className={classes.grow} />
            <Button className={classes.button} onClick={this.handleNoteCurrentPage}>
              {currentPage === 'notes' ? 'Notes' : 'View Notes'}
            </Button>
            <Button className={classes.button} onClick={this.handleRemCurrentPage}>
            {currentPage === 'rem' ? 'Remainders' : 'View Remainders'}
            </Button>
            <Button className={classes.button} onClick={isLoggedIn ? this.handleLogout : this.handleLogin}>
              {isLoggedIn ? 'Logout' : 'Login'}
            </Button>
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

Header.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Header);    // HOC -> Higher Order Component which takes another compoennt as an argument
