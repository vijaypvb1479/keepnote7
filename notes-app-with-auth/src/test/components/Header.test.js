import React from 'react';
import {mount } from 'enzyme';
import Header from '../../components/Header';

import {createSerializer} from 'enzyme-to-json';
 
expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

test('test Header', () => {

    const wrapper = mount(<Header />);
    expect(wrapper).toMatchSnapshot();


});