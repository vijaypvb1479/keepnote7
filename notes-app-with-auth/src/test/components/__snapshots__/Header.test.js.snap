// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Header 1`] = `
<div
  className="Header-root-1"
>
  <header
    className="MuiPaper-root-14 MuiPaper-elevation4-20 MuiAppBar-root-5 MuiAppBar-positionStatic-9 MuiAppBar-colorPrimary-12"
  >
    <div
      className="MuiToolbar-root-41 MuiToolbar-regular-43 MuiToolbar-gutters-42"
    >
      <h6
        className="MuiTypography-root-45 MuiTypography-h6-62 MuiTypography-colorInherit-74 MuiTypography-noWrap-71 Header-title-3"
      >
        Notes
      </h6>
      <div
        className="SearchBar-search-80"
      >
        <div
          className="SearchBar-searchIcon-81"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-84"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </div>
        <div
          className="MuiInputBase-root-93 SearchBar-inputRoot-82"
          onClick={[Function]}
        >
          <input
            className="MuiInputBase-input-103 SearchBar-inputInput-83"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Searchâ€¦"
            type="text"
          />
        </div>
      </div>
      <div
        className="Header-grow-4"
      />
      <button
        className="MuiButtonBase-root-136 MuiButton-root-110 MuiButton-text-112 MuiButton-flat-115"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-111"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-139"
        />
      </button>
    </div>
  </header>
</div>
`;
